src/main/java/frc/robot/Robot.java

```diff
@@ -47,11 +47,15 @@ public void robotInit() {
     m_controller = new CommandJoystick(0);
     m_swerve = new SwerveDriveTrainSubsystem();
 
-    m_controller.button(3).onTrue(Commands.run(() -> {
-      m_swerve.resetGyro();
-      DriverStation.reportWarning("Y button pressed", false);
+    m_controller.button(3).onTrue(Commands.runOnce(() -> {
+     m_swerve.resetGyro();
+     
+     DriverStation.reportWarning("Y button pressed", false);
```

Comment:
Now that you've verified that the command works properly, you no longer need to report this information to the Driver Station. With that being said, it is important for us to know that the reset occurred and when in case we need to refer back to logs. 

For now, I suggest logging this event to the built-in console logger:
```suggestion
     System.out.printLn("[DRIVE] Reset gyro");
```